metabase/metabase
# Switch the collection page to table-based layout, display last edit info

Switches collection items layout from list-based to table-based one. Apart from item type and name, the table also shows who last edited a thing and when. Closes #6750

Important notes:

- "Last edited at" and "Last edited by" columns are not displayed for Pulses. We retrieve this info from the most recent revision, and Pulses don't have them
- drag-n-drop for collection items for table layout is going to be added in the next PR
- this PR also removes code and tests related to collection items filtering, as it's going to be replaced with items sorting

Refactorings:

* Split [EntityItem](https://github.com/metabase/metabase/blob/fb3b85f757ba8eeba87e1b114155e93ce6f96c65/frontend/src/metabase/components/EntityItem.jsx) into a few components (icon, name, action menu), so they can be rendered independently in different table cells

### To Verify

1. Open any collection (e.g. `/collection/root`)
2. Check out collections that: are empty, have no pinned items, have only pinned items, have both pinned and unpinned items

### Screenshots

**Before**

<img width="1095" alt="before" src="https://user-images.githubusercontent.com/17258145/120801416-cb5baf00-c549-11eb-9278-853f5bd40c03.png">

**After**

<img width="1439" alt="after" src="https://user-images.githubusercontent.com/17258145/120801389-c434a100-c549-11eb-9f14-7df8d4325cf2.png">
